src/lib/oracle/index.ts

```diff
@@ -0,0 +1,89 @@
+import axios from "axios";
+import { BigNumber } from "ethers";
+
+import { GELATO_RELAY_URL } from "../../constants";
+import { getHttpErrorMessage } from "../../utils";
+
+/**
+ * @param {number} chainId - Chain Id
+ * @returns {Promise<boolean>} Boolean to demonstrate if the oracle is active on the provided chain
+ *
+ */
+export const isOracleActive = async (chainId: number): Promise<boolean> => {
+  const oracles = await getGelatoOracles();
+  return oracles.includes(chainId.toString());
+};
+
+/**
+ * @returns {Promise<string[]>} List of chain ids where the Gelato Oracle is active
+ *
+ */
+export const getGelatoOracles = async (): Promise<string[]> => {
+  try {
+    return (await axios.get(`${GELATO_RELAY_URL}/oracles/`)).data.oracles;
+  } catch (error) {
+    throw new Error(
+      `GelatoRelaySDK/getGelatoOracles: Failed with error: ${getHttpErrorMessage(
+        error
+      )}`
+    );
+  }
+};
+
+/**
+ * @param {number} chainId - Chain Id
+ * @returns {Promise<string[]>} List of all payment tokens on the provided chain
+ *
+ */
+export const getPaymentTokens = async (chainId: number): Promise<string[]> => {
+  try {
+    return (
+      await axios.get(`${GELATO_RELAY_URL}/oracles/${chainId}/paymentTokens/`)
+    ).data.paymentTokens;
+  } catch (error) {
+    throw new Error(
+      `GelatoRelaySDK/getPaymentTokens: Failed with error: ${getHttpErrorMessage(
+        error
+      )}`
+    );
+  }
+};
+
+/**
+ * @param {number} chainId - Chain Id
+ * @param {string} paymentToken - Payment Token
+ * @param {BigNumber} gasLimit - Gas Limit
+ * @param {boolean} isHighPriority - Priority Level
+ * @param {BigNumber} [gasLimitL1=BigNumber.from(0)] - Gas Limit for Layer 1
+ * @returns {Promise<BigNumber>} Estimated Fee
+ *
+ */
+export const getEstimatedFee = async (
+  chainId: number,
+  paymentToken: string,
+  gasLimit: BigNumber,
+  isHighPriority: boolean,
```

Comment:
I think we should remove this `isHighPriority`  params because as far as I remember, it's not doing anything and just creates confusion

Reply:
I just tested it, it returns different results based on the selected priority. The current result is `lowPrioFee * 5 = highPrioFee`